src/docdb/commands.ts

```diff
@@ -118,8 +119,18 @@ export class DocDBCommands {
         return undefined;
     }
 
+    private static validateDocumentID(name: string): string | undefined | null {
+        if (/^[-a-z0-9]+$/ig.test(name)) {
+            return "ID should contain only the following : digits, alphabets and -";
+        }
+        if (name.length > 40) {
```

Comment:
How did you get a limit of 40? I successfully created a docId of length 100

Reply:
I'll admit 40 was an estimate on the length of the hash it generated (including hyphens). I didn't test it completely. I'll remove the length check if I can't quickly determine a tight upper bound.